programs/zstdcli.c

```diff
@@ -452,6 +465,38 @@ static void printVersion(void)
 #endif
 }
 
+/* Environment variables for parameter setting */
+#define ENV_CLEVEL "ZSTD_CLEVEL"
+
+/* functions that pick up environment variables */
+int init_cLevel(void) {
+    const char* const env = getenv(ENV_CLEVEL);
+    if (env) {
+        const char *ptr = env;
+        int sign = 1;
+        if (*ptr == '-') {
+            sign = -1;
+            ptr++;
+        } else if (*ptr == '+') {
+            ptr++;
+        }
+
+        if ((*ptr>='0') && (*ptr<='9')) {
```

Comment:
That's a good idea! It requires a bit more refactoring work as `readU32FromChar()` is used in many places under slightly different contexts.

Reply:
We are too close to a release to modify any behavior, so no need to refactor.

Also, I suspect that the `if ((*ptr>='0') && (*ptr<='9'))` test is extraneous,
because it's already included in `readU32FromCharChecked()`, 
so the function will fail and return `1` if the first character does not respect this condition anyway.

Reply:
readU32FromCharChecked() in its current form is pretty bullet-proof, and it will return 0 as the value read if the first char is not a numeral. This feature has implications on allowed command line option format, e.g., currently,
`$ZSTD -bi0 --fast tmp1`
is equivalent to
`$ZSTD -bi --fast tmp1`.

If we remove `if ((*ptr>='0') && (*ptr<='9'))` in `int_cLevel(...)`, setting `ZSTD_CLEVEL` to an empty string will be considered valid and interpreted as ZSTD_CLEVEL=0. 

Also the same check appears in current release: (zstdcli.c, line 754)
```
#ifndef ZSTD_NOCOMPRESS
                    /* compression Level */
                    if ((*argument>='0') && (*argument<='9')) {
                        dictCLevel = cLevel = readU32FromChar(&argument);
                        continue;
                    }
#endif
```
which is also necessary.

In the future, we can spend some time rationalizing the acceptable option format for the command line. Also it would be helpful to add an option to display all the parameter settings actually used by our compression/decompression algorithms.